packages/preferences/src/browser/views/preference-editor-widget.ts

```diff
@@ -314,18 +317,26 @@ export class PreferencesEditorWidget extends BaseWidget implements StatefulWidge
                     if (renderer?.visible) {
                         // When filtered, treat the first visible child as the selected node, since it will be the one scrolled to.
                         this.lastUserSelection = renderer.nodeId;
-                        renderer.node.scrollIntoView();
+                        this.scrollWithoutModelUpdate(renderer.node);
                         return;
                     }
                 }
             } else {
                 const { id, collection } = this.analyzeIDAndGetRendererGroup(node.id);
                 const renderer = collection.get(id);
-                renderer?.node.scrollIntoView();
+                this.scrollWithoutModelUpdate(renderer?.node);
             }
         }
     }
 
+    /** Ensures that we don't set the model's selection while attempting to scroll in reaction to a model selection change. */
+    protected scrollWithoutModelUpdate(node?: HTMLElement): void {
+        if (!node) { return; }
+        this.shouldUpdateModelSelection = false;
+        node.scrollIntoView();
+        requestAnimationFrame(() => this.shouldUpdateModelSelection = true);
```

Comment:
Two comments here: Why are we delaying this here? If it's for the code to execute after some promises we create in the meantime, that's not reliable because execution order of promises vs requestAnimationFrame() is not guarantueed.
Second: are we sure this code is not invoked before the Window is visible? Because otherwise, this might delay loading, see https://github.com/eclipse-theia/theia/issues/13798

Reply:
It appears that `setTimeout` does not reliably push the reset past the running of the handler. This particular `requestAnimationFrame` currently doesn't run at startup, as we don't restore model selection independently.